packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts

```diff
@@ -85,6 +85,15 @@ export interface NetworkLoadBalancedFargateServiceProps extends NetworkLoadBalan
    * @default Latest
    */
   readonly platformVersion?: FargatePlatformVersion;
+
+  /**
+   * The health check settings to use when evaluating the health of your EC2 instances.
+   * It is used for determining if a service is healthy and consists at least of an array of commands
+   * which are run when a health check is triggered.
+   *
+   * @default - Health check configuration from container.
+   */
+  readonly healthCheck?: HealthCheck;
```

Comment:
After thinking about this more, I'm not sure we should expose this on the top level props. For an ECS service there behind a load balancer there are two different types of health checks that you can use.

1. ELB health checks (configured on the NLB in this case). The NLB will make a request to the service on a specific path/port and expect a certain response.
2. Container health checks (what this PR exposes). The service itself can perform health checks against the running containers (in _addition_ to the ELB health checks).

I'm not sure this is a common use case for services running behind an ELB. The other linked PR that implemented this for the QueueProcessing service makes more sense because it is a worker service that is not running behind an ELB so container health checks are the _only_ type of health check available.

I think in the majority of cases the user wants to configure the ELB health check and exposing this at the top level (an not the elb health check) would lead to more confusion and users configuring container health checks when they don't need to.

cc @madeline-k for a second opinion.

Reply:
@corymhall Ok I think this makes sense. Maybe we should close this (and the linked issue) and come up with a solution if there is a real need for this. And even then explicitly document the consequences.

And I want to apologize for implementing this feature without really thinking about it. I will try to go more *in-depth* in a topic I will implement in the future.